site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -355,229 +355,236 @@ public function ajaxUploadCourseMaterialsFiles(): JsonResponse {
+                                foreach ($dirs as $dir) {
+                                    for ($i = $count; $i < $j + $count; $i++) {
+                                        if (!isset($dirs_to_make[$i])) {
+                                            $dirs_to_make[$i] = $upload_path . '/' . $dir;
+                                        }
+                                        else {
+                                            $dirs_to_make[$i] .= '/' . $dir;
+                                        }
                                     }
+                                    $j--;
                                 }
-                                $j--;
+                                $index++;
                             }
-                            $index++;
-                        }
-                    }
-                    else {
-                        if (!@copy($uploaded_files[1]["tmp_name"][$j], $dst)) {
-                            return JsonResponse::getErrorResponse("Failed to copy uploaded file {$uploaded_files[1]['name'][$j]} to current location.");
                         }
                         else {
-                            $details['type'][$index] = $is_external_link_file ? CourseMaterial::LINK : CourseMaterial::FILE;
-                            $details['path'][$index] = $dst;
-                            $index++;
+                            if (!@copy($uploaded_files[1]["tmp_name"][$j], $dst)) {
+                                return JsonResponse::getErrorResponse("Failed to copy uploaded file {$uploaded_files[1]['name'][$j]} to current location.");
+                            }
+                            else {
+                                $details['type'][$index] = CourseMaterial::FILE;
+                                $details['path'][$index] = $dst;
+                                $index++;
+                            }
                         }
                     }
-                }
-                else {
-                    return JsonResponse::getErrorResponse("The tmp file '{$uploaded_files[1]['name'][$j]}' was not properly uploaded.");
-                }
-                // Is this really an error we should fail on?
-                if (!@unlink($uploaded_files[1]["tmp_name"][$j])) {
-                    return JsonResponse::getErrorResponse("Failed to delete the uploaded file {$uploaded_files[1]['name'][$j]} from temporary storage.");
+                    else {
+                        return JsonResponse::getErrorResponse("The tmp file '{$uploaded_files[1]['name'][$j]}' was not properly uploaded.");
+                    }
+                    // Is this really an error we should fail on?
+                    if (!@unlink($uploaded_files[1]["tmp_name"][$j])) {
+                        return JsonResponse::getErrorResponse("Failed to delete the uploaded file {$uploaded_files[1]['name'][$j]} from temporary storage.");
```

Comment:
My opinion is that perhaps this isn't something we should fail on, but instead hide from the user. Perhaps this is instead something the sys admin should be concerned with since the user will have very limited information on why this is failing.